From 8ac6320697a640db0be6e8cda9f69eb20e515e65 Mon Sep 17 00:00:00 2001
From: Adrien Destugues <pulkomandy@pulkomandy.tk>
Date: Sun, 8 Oct 2017 10:16:27 +0200
Subject: Fix crash

Not sure why, but this ends up being called with be_control_look NULL
now.

diff --git a/src-beam/BmToolbarButton.cpp b/src-beam/BmToolbarButton.cpp
index 490d5f9..4aa6d23 100644
--- a/src-beam/BmToolbarButton.cpp
+++ b/src-beam/BmToolbarButton.cpp
@@ -236,7 +236,7 @@ void BmToolbarButton::Draw( BRect updateRect) {
 
 	BRect rect(Bounds());
 #ifdef __HAIKU__
-	if (mHighlighted && IsEnabled()) {
+	if (be_control_look && mHighlighted && IsEnabled()) {
 		// draw higlighting border
 		uint32 flags = 0;
 		if (Value())
-- 
2.14.2


From 07bc84afb7f18dc3e91215d11fb6fc2bbabe24e2 Mon Sep 17 00:00:00 2001
From: Adrien Destugues <pulkomandy@pulkomandy.tk>
Date: Sun, 8 Oct 2017 11:25:09 +0200
Subject: Fix build with current flex

It is confused if we are C89 but define INT32_MAX anyway. So pretend we
are C99.

diff --git a/src-filter-addons/src-sieve/src-libSieve/Jamfile b/src-filter-addons/src-sieve/src-libSieve/Jamfile
index cbe5db4..884d007 100644
--- a/src-filter-addons/src-sieve/src-libSieve/Jamfile
+++ b/src-filter-addons/src-sieve/src-libSieve/Jamfile
@@ -11,6 +11,8 @@ if $(OSPLAT) = X86 {
 	CCFLAGS += -proto lax -w nounwanted -w nounused ;
 }
 
+CFLAGS += --std=gnu9x ;
+
 SubDirSysHdrs $(COMMON_FOLDER)/include ;
 
 # <pe-src>
-- 
2.14.2

